icu4j/main/collate/src/test/java/com/ibm/icu/dev/test/collator/CollationAPITest.java

```diff
@@ -530,8 +530,7 @@ public void TestJunkCollator(){
 
         String colrules = ((RuleBasedCollator)col).getRules();
         String junkrules = ((RuleBasedCollator)junk).getRules();
-        doAssert(colrules == junkrules || colrules.equals(junkrules),
-                   "The default collation should be returned.");
+        doAssert(colrules.equals(junkrules), "The default collation should be returned.");
```

Comment:
FYI -- looks like that was an optimization which hopefully lives in the .equals() impl as well anyway

Reply:
From https://errorprone.info/bugpattern/ReferenceEquality:

* Most .equals (should) have a "fast path"
  ```java
  @Override
  public boolean equals(Object other) {
    if (this == other) {
      return true; // fast path, reference equality is allowed here
    }
    ...
  ```

Optional, I can replace with `assertSame(b, a)`
```java
assertSame(colrules, junkrules); // this uses ==
assertEquals(colrules, junkrules); // this uses .equals
```

Although in the original `if` it is an OR, and what I propose is an AND.

But basically having it an OR is too loose.
Even if `colrules != junkrules` the `colrules.equals(junkrules)` might succeed.
So my testing is not that good, I think.

Reply:
I've replaced it with `assertSame(colrules, junkrules)` only.

There is no point to also test `assertEquals`, these are strings, and we know that `equals` is implemented correctly.
If `a == b` then `a.equals(b)` and `b.equals(a)`.
If that would fail for `String` it would be hilarious :-)

Reply:
I actually thought that "equals" was fine here, just adding a comment that helped justify it. But if "same" works, that's ok too.